react/react.d.ts

```diff
@@ -125,7 +125,7 @@ declare namespace __React {
         state: S;
         context: {};
         refs: {
-            [key: string]: ReactInstance
+            [key: string]: Component<any, any> & Element
```

Comment:
this is wrong; it should not be an intersection type; union was the right type. you have to narrow the type from `Component<any, any> | Element` to just `Element` to get your code to work